test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -5078,6 +5078,104 @@ BOOST_AUTO_TEST_CASE(invalid_address_length)
 	CHECK_WARNING(text, "checksum");
 }
 
+BOOST_AUTO_TEST_CASE(interface)
+{
+	char const* text = R"(
+		interface I {
+		}
+	)";
+	success(text);
+}
+
+BOOST_AUTO_TEST_CASE(interface_constructor)
+{
+	char const* text = R"(
+		interface I {
+			function I();
+		}
+	)";
+	success(text);
```

Comment:
I would say this should signal failure. In general, a constructor has to be implemented, regardless of whether it is used in an interface or not (I just tried it, it causes an internal compiler error).